### PR TITLE
Fix: looker_saml_config - pass an id in the request body

### PR DESCRIPTION
**Pull request by jerny-stoiclane**
Hello!  First wanted to say thank you for this provider.  

I am using the `looker_saml_config` settings but get errors when adding a new block for the `groups_with_role_ids` attribute.  Below is the error

```bash
╷
│ Error: validation error on fields: 'saml_group' (Can not include 'looker_group_name' field without 'id' field.)
│
│   with looker_saml_config.okta,
│   on looker_saml.tf line 1, in resource "looker_saml_config" "okta":
│    1: resource "looker_saml_config" "okta" {
│
╵
Releasing state lock. This may take a few moments...
```
The api expects the `Id` field to be present even if its an empty value.  So here is an attempt at a fix.

### CODE REVIEW

**Review by jessicasomaiya — COMMENTED** [review]
Hello! Thank you for highlighting this bug and making a PR to fix it. I’ve come across this bug too and as you mentioned, the api is expecting the Id field in the request.

The strange thing is that this group Id is never used as Looker created a new group as a result of the API call with a new Id , discarding the Id we supply. This is ultimately a bug in the Looker API imo but a good workaround is to supply any group Id just to satisfy the request requirements. 

To follow conventions that are already used in the provider [(see here)](https://github.com/resolutionlife/terraform-provider-looker/blob/main/looker/resource_looker_saml_config.go#L380), I’d suggest using a random integer as the Id rather than using the hash of the Looker group and the SAML group name.

File: looker/resource_looker_saml_config.go
Comment (jessicasomaiya) [inline looker/resource_looker_saml_config.go:309]: Looks great! There's just one too may brackets here. 
```suggestion
				id := fmt.Sprintf("%d", rand.Int())
```

Comment (jerny-stoiclane) [reply looker/resource_looker_saml_config.go:309]: Doh, done before coffee

Comment (jerny-stoiclane) [reply looker/resource_looker_saml_config.go]: Updated

**Comment by jessicasomaiya** [comment]
I'm just looking into these acceptance test failures

**Comment by jessicasomaiya** [comment]
Hey @jerny-stoiclane! Good news, we've just merged a change that should fix the acceptance tests. The issue was to do with accessing environment variables on this repo. We're now using dummy env vars in our test suite. 

To update your branch with this new change, please sync your fork and then merge the new changes into your branch. This should do the trick!

**Comment by jerny-stoiclane** [comment]
@jessicasomaiya nice!  No problem I can do that.

**Comment by jerny-stoiclane** [comment]
Hi @ponteilla.  I should have rebased the main branch changes onto the branch in the PR.  Do I need to do something else to get the https://github.com/resolutionlife/terraform-provider-looker/actions/runs/4386622795 to trigger?  It looks like it is still waiting for approval.

**Review by jessicasomaiya — APPROVED** [review]
Thanks again for contributing, looks great!

**Comment by jessicasomaiya** [comment]
@jerny-stoiclane I just merger your PR and released the provider so your change is in `v0.2.3`. Best of luck!

**Comment by jerny-stoiclane** [comment]
@jessicasomaiya awesome thanks!

**Pull request merged**
